lib/github_fetch/github_fetch.go

```
@@ -79,10 +88,32 @@ func (gf *GitHubFetcher) Get(parentSpan tracer.Span, owner string, repo string,
 	if url == nil {
 		return nil, fmt.Errorf("url is nil")
 	}
-	return gf.getArchive(url)
+	return gf.getArchive(url, excludes)
 }
 
-func (gf *GitHubFetcher) getArchive(archiveURL *url.URL) (io.Reader, error) {
+// parseDockerIgnoreIfExists will parse the docker ignore file if it exists in order to determine which patterns should be excluded.
+// The excluded patterns are intended to be used with a pattern matcher.
+func (gf *GitHubFetcher) parseDockerIgnoreIfExists(ctx context.Context, owner, repo string, opt *github.RepositoryContentGetOptions) ([]string, error) {
+	fc, _, resp, err := gf.c.Repositories.GetContents(ctx, owner, repo, dockerIgnorePath, opt)
+	if err != nil {
+		if resp.StatusCode == 404 {
+			// Not all repos will have dockerignore, just move along
+			return []string{}, nil
+		}
+		return nil, fmt.Errorf("error getting .dockerignore: %v", err)
+	}
+	content, err := fc.GetContent()
+	if err != nil {
+		return nil, fmt.Errorf("error getting content from %v, %v", dockerIgnorePath, err)
+	}
+	excludes, err := dockerignore.ReadAll(strings.NewReader(content))
+	if err != nil {
+		return nil, fmt.Errorf("error parsing %v, %v", dockerIgnorePath, err)
```

Do we want to fail on an invalid dockerignore, or just ignore? What does `docker build` do?

---

Good question. 

Just tested this locally and `docker build` fails with an invalid dockerignore. 


```
$ docker build .
error checking context: 'syntax error in pattern'.
```

I think it would be ideal to be as close to the docker CLI's behavior as possible.